web/src/stream_data.ts

```diff
@@ -619,14 +619,28 @@ function can_administer_channel(sub: StreamSubscription): boolean {
     );
 }
 
+export function can_unsubscribe(sub: StreamSubscription): boolean {
+    if (can_unsubscribe_others(sub)) {
```

Comment:
Might be good to add a comment here explaining that `can_unsubscribe_others` handles checking `can_administer_channel_group` setting.